diff --git a/src/communication/UdpDriver.cpp b/src/communication/UdpDriver.cpp
index 4d65e53..9585e76 100644
--- a/src/communication/UdpDriver.cpp
+++ b/src/communication/UdpDriver.cpp
@@ -88,7 +88,7 @@ bool UdpDriver::isSameSubnet(ip4_addr_t addr) {
 }
 
 bool UdpDriver::isMulticastAddress(ip4_addr_t addr) {
-  return ((addr.addr >> 28) == 14);
+  return ip4_addr_ismulticast(&addr);
 }
 
 bool UdpDriver::joinMultiCastGroup(ip4_addr_t addr) const {
diff --git a/src/discovery/ParticipantProxyData.cpp b/src/discovery/ParticipantProxyData.cpp
index 24703fa..232e674 100644
--- a/src/discovery/ParticipantProxyData.cpp
+++ b/src/discovery/ParticipantProxyData.cpp
@@ -170,8 +170,8 @@ bool ParticipantProxyData::readLocatorIntoList(
   for (auto &proxy_locator : list) {
     if (!proxy_locator.isValid()) {
       bool ret = full_length_locator.readFromUcdrBuffer(buffer);
-      if (ret && full_length_locator.isSameSubnet() ||
-          full_length_locator.isMulticastAddress()) {
+      if (ret && (full_length_locator.isSameSubnet() ||
+                  full_length_locator.isMulticastAddress())) {
         proxy_locator = LocatorIPv4(full_length_locator);
         SPDP_LOG("Adding locator: %u %u %u %u \n",
                  (int)proxy_locator.address[0], (int)proxy_locator.address[1],
