diff --git a/src/discovery/SPDPAgent.cpp b/src/discovery/SPDPAgent.cpp
index 98eb0ed..6ba9f7b 100644
--- a/src/discovery/SPDPAgent.cpp
+++ b/src/discovery/SPDPAgent.cpp
@@ -165,6 +165,8 @@ void SPDPAgent::processProxyData() {
   if (mp_participant->addNewRemoteParticipant(m_proxyDataBuffer)) {
     addProxiesForBuiltInEndpoints();
     m_buildInEndpoints.spdpWriter->setAllChangesToUnsent();
+    m_buildInEndpoints.sedpPubWriter->setAllChangesToUnsent();
+    m_buildInEndpoints.sedpSubWriter->setAllChangesToUnsent();
 #if SPDP_VERBOSE && RTPS_GLOBAL_VERBOSE
     SPDP_LOG("Added new participant with guid: ");
     printGuidPrefix(m_proxyDataBuffer.m_guid.prefix);
